test/test_torch.py

```diff
@@ -6050,7 +6053,9 @@ def test_inverse_helper(matrix, batches, n):
                 for mat in matrix.contiguous().view(-1, n, n):
                     expected_inv_list.append(torch.inverse(mat))
                 expected_inv = torch.stack(expected_inv_list).view(*batches, n, n)
-                self.assertEqual(matrix_inverse, expected_inv)
+                # single-inverse is done using MAGMA, while batched inverse is done using cuBLAS
+                # individual values can be significantly different, hence higher atol is used
+                self.assertEqual(matrix_inverse, expected_inv, atol=1e-1, rtol=0)
```

Comment:
I think currently single-inverse is on cusolver, but batched inverse is mostly on MAGMA. cublas batched inverse performance is too bad, and is not used.

Different implementations shouldn't give these large differences. Can you try running tests only in double precision, and see if the numerical difference is still this large?

Reply:
>I think currently single-inverse is on cusolver, but batched inverse is mostly on MAGMA.

Indeed, I mixed them up.

I will check the tolerances.

Reply:
Double precision is fine and tests pass for low tolerance (`atol=1e-11` in my case).
But for floats, individual entries can vary significantly:
>Tensors failed to compare as equal! With rtol=0 and atol=0.1, found 2553 element(s) (out of 393216) whose difference(s) exceeded the margin of error (including 0 nan comparisons). The greatest difference was 0.37847900390625 (820.0460815429688 vs. 819.6676025390625), which occurred at index (1, 1, 254, 71).


I think MAGMA and cuSOLVER can give these large differences as long as the `identity == inv(A) A` tests pass
```python
self.assertEqual(identity.expand_as(matrix), torch.matmul(matrix, matrix_inverse))
self.assertEqual(identity.expand_as(matrix), torch.matmul(matrix_inverse, matrix))
```